src/main/java/task/TaskList.java

```diff
@@ -0,0 +1,71 @@
+package task;
+
+import components.Parser;
+import exception.NyabotParseException;
+
+import java.util.ArrayList;
+
+
+public class TaskList {
+    private ArrayList<Task> tasks = new ArrayList<Task>();
+
+    public void addTask(Task task) {
+        tasks.add(task);
+    }
+
+    public void replaceTasks(TaskList taskList) {
+        this.tasks = taskList.tasks;
+    }
+
+    public void deleteTask(Task task) {
+        tasks.remove(task);
+    }
+    public void deleteTask(int taskNumber) {
+
+        tasks.remove(taskNumber);
+    }
+
+    public String displayTasks() {
+        StringBuilder textListString = new StringBuilder();
+        for (int i = 0; i < tasks.size(); i++) {
+            textListString.append((i + 1)).append(". ").append(tasks.get(i));
+            if (i < tasks.size()-1) {
+                textListString.append("\n");
+            }
+
+        }
+        return textListString.toString();
+    }
+
+    public String displayTasksSaveable() throws NyabotParseException {
```

Comment:
Perhaps you could rename this method? Currently, it doesn't seem that the method displays the strings, but instead returns them, so "display" may not be a suitable term for this method.